security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java

```diff
@@ -376,18 +388,23 @@ public void handleMessage(SoapMessage message) throws Fault {
     }
 
     private void createSecurityToken(SoapVersion version, SOAPFactory soapFactory, SOAPElement securityHeader) {
-        AnonymousAuthenticationToken token = new AnonymousAuthenticationToken(BaseAuthenticationToken.DEFAULT_REALM);
-
-        //synchronize the step of requesting the assertion, it is not thread safe
-        Subject subject = null;
-        synchronized (lock) {
-            try {
-                subject = securityManager.getSubject(token);
-            } catch (SecurityServiceException sse) {
-                LOGGER.warn("Unable to request subject for anonymous user.", sse);
+        Subject subject = anonymousSubject;
```

Comment:
I am not sure how reusable the code would be outside of this class.  I can extract the subject logic to make this function more readable.